From 07d9a86108b60f6c991cebde9b992fcfa94b7fd5 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@freescale.com>
Date: Thu, 29 Oct 2015 17:37:51 +0800
Subject: [PATCH 0814/5242] MLK-11790 dts: imx6: fix wlreg_on regulator always
 on issue

commit  d25bc30a79b223e422689b5687f8c0f0dd9e3958 from
https://source.codeaurora.org/external/imx/linux-imx.git

wlreg_on regulator always on will make WiFi module unable to do reset
during rmmod, then the re-insmod will fail.

In stead of make wlreg_on always on, we make vmmc regulator always on
for BT to work properly.
Then WiFi can still function well on wlreg_on regulator enable/disable.

Fixes: commit f17b5e5be (dts: mx6ul evk: set regulator wlreg_on always on)
Signed-off-by: Dong Aisheng <aisheng.dong@freescale.com>
(cherry picked from commit e92b727494b10c7a13e9491f55fc07a7ed7622a3)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx-sdb-btwifi.dts  |    5 ++++-
 arch/arm/boot/dts/imx6ul-evk-btwifi.dtsi |    5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-sdb-btwifi.dts b/arch/arm/boot/dts/imx6sx-sdb-btwifi.dts
index 575ed3c..a74d274 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-btwifi.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-btwifi.dts
@@ -24,7 +24,6 @@
 			regulator-name = "wlreg_on";
 			gpio = <&gpio6 10 0>;
 			startup-delay-us = <100>;
-			regulator-always-on;
 			enable-active-high;
 		};
 	};
@@ -93,6 +92,10 @@
 	bus-width = <1>;
 };
 
+&vcc_sd3 {
+	regulator-always-on;
+};
+
 &usdhc3 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_wifi>;
diff --git a/arch/arm/boot/dts/imx6ul-evk-btwifi.dtsi b/arch/arm/boot/dts/imx6ul-evk-btwifi.dtsi
index 1f3f8d5..99a67cf8 100644
--- a/arch/arm/boot/dts/imx6ul-evk-btwifi.dtsi
+++ b/arch/arm/boot/dts/imx6ul-evk-btwifi.dtsi
@@ -21,7 +21,6 @@
 			regulator-name = "wlreg_on";
 			gpio = <&gpio5 1 0>;
 			startup-delay-us = <100>;
-			regulator-always-on;
 			enable-active-high;
 		};
 	};
@@ -42,6 +41,10 @@
 	};
 };
 
+&reg_sd1_vmmc {
+	regulator-always-on;
+};
+
 &usdhc1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc1 &pinctrl_wifi>;
-- 
1.7.9.5

